KEP-5999:Documentation updates for alpha release (#56323)
kubernetes/website: content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

In releases after v1.13, local HTTP proxy environment variable settings do not
affect the HTTP liveness probe.

### Use HTTP/2 cleartext (h2c) with HTTP probes {#use-h2c-with-http-probes}

{{< feature-state for_k8s_version="v1.37" state="alpha" >}}

> By default the kubelet sends HTTP/1.1 requests when executing an httpGet probe.

Reviewer: ```suggestion
By default the kubelet sends HTTP/1.1 requests when executing an HTTP probe.
```

Author: fixed, thanks

If your application serves health endpoints only over HTTP/2 cleartext (h2c),

> you can set the `protocol` field to `HTTP2`. This requires the

Reviewer: ```suggestion
you can add the `protocol` field to the `httpGet` field in the probe specification and specify
a value of `HTTP2`. This requires the
```

Author: fixed, thanks

`H2CContainerProbe`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be

> enabled on both the kube-apiserver and the kubelet.

Reviewer: nit: use code formatting for `kubelet` and `kube-apiserver`

Author: fixed .

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: h2c-app
spec:
  containers:
  - name: app
    image: my-h2c-app:latest
    ports:
    - containerPort: 8080
    livenessProbe:
      httpGet:
        path: /healthz
        port: 8080
        protocol: HTTP2
      initialDelaySeconds: 5
      periodSeconds: 10
```

When `protocol` is set to `HTTP2`, the kubelet connects using HTTP/2 cleartext
(h2c) — HTTP/2 over plain TCP without TLS. You cannot combine `protocol: HTTP2`
with `scheme: HTTPS`; the API server rejects that combination. The `host` field

> must also be empty when using `protocol: HTTP2`.

Reviewer: ```suggestion
(h2c) — HTTP/2 over plain TCP without TLS. The following configurations aren't
supported:

 *    `protocol: HTTP2` and `scheme: HTTPS`
 *    `protocol: HTTP2` and a value in the `host` field
```

Author: done.

> When the feature gate is disabled, the API server silently drops the `protocol`

Reviewer: If the feature gate is disabled, the API server removes the `protocol` field from new or updated Pods.

Author: fixed.

In releases after v1.13, local HTTP proxy environment variable settings do not
affect the HTTP liveness probe.

### Use HTTP/2 cleartext (h2c) with HTTP probes {#use-h2c-with-http-probes}

> {{< feature-state for_k8s_version="v1.37" state="alpha" >}}

Reviewer 2: Use `feature_gate_name` instead of hardcoding version and state — it auto-updates when the feature graduates, and matches the concept page (probes.md line 386).

```suggestion
{{< feature-state feature_gate_name="H2CContainerProbe" >}}
```

Author: added

> By default the kubelet sends HTTP/1.1 requests when executing an HTTP probe.

Reviewer 2: `kubelet` should have backticks for consistency

```suggestion
By default the `kubelet` sends HTTP/1.1 requests when executing an HTTP probe.
```

Author: fixed

If your application serves health endpoints only over HTTP/2 cleartext (h2c),
you can add the `protocol` field to the `httpGet` field in the probe specification and
specify a value of `HTTP2`. This requires the `H2CContainerProbe`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be
enabled on both the `kube-apiserver` and the `kubelet`.

> ```yaml

Reviewer 2: This is a complete manifest. The rest of this page uses `{{% code_sample %}}` for manifests (for example, `pods/probe/exec-liveness.yaml`, `pods/probe/http-liveness.yaml`). Extract this YAML to a file like `content/en/examples/pods/probe/h2c-liveness.yaml` and reference it with:

```markdown
{{% code_sample file="pods/probe/h2c-liveness.yaml" %}}
```

Author: done, thanks

apiVersion: v1
kind: Pod
metadata:
  name: h2c-app
spec:
  containers:
  - name: app

>     image: my-h2c-app:latest

Reviewer 2: `my-h2c-app:latest` is a placeholder image that does not exist — readers cannot `kubectl apply` this manifest. Tru using workable image or at least add a comment

```suggestion
    image: my-h2c-app:latest  # replace with an image that serves h2c health endpoints
```

Author: done

    ports:
    - containerPort: 8080
    livenessProbe:
      httpGet:
        path: /healthz
        port: 8080
        protocol: HTTP2
      initialDelaySeconds: 5
      periodSeconds: 10
```

> When `protocol` is set to `HTTP2`, the kubelet connects using HTTP/2 cleartext

Reviewer 2: ```suggestion
When `protocol` is set to `HTTP2`, the `kubelet` connects using HTTP/2 cleartext
```

Author: done, thanks
